jobs/ror/ProcessRorCsv.php

```diff
@@ -0,0 +1,287 @@
+    {
+        $ror = $row[$this->dataMappingIndex['ror']];
+        $displayLocale = (!empty($row[$this->dataMappingIndex['displayLocale']]))
+            ? $row[$this->dataMappingIndex['displayLocale']]
+            : $this->noLocale;
+        $isActive = (strtolower($row[$this->dataMappingIndex['isActive']]) === 'active') ? 1 : 0;
+        $searchPhrase = $ror;
+
+        $namesIn = $row[$this->dataMappingIndex['names']];
+        $namesOut = [];
+        if (!empty($namesIn)) {
+            $names = array_map('trim', explode(';', $namesIn));
+            for ($i = 0; $i < count($names); $i++) {
+                $name = array_map('trim', explode(':', $names[$i]));
+                if (count($name) === 2) {
+                    $namesOut[$name[0]] = trim($name[1]);
+                    $searchPhrase .= ' ' . $namesOut[$name[0]];
+                }
+            }
+        }
+
+        if (empty($namesOut[$displayLocale])) {
+            $namesOut[$displayLocale] = $row[$this->dataMappingIndex['displayName']];
+        }
+
+        return [
+            'ror' => $ror,
+            'displayLocale' => $displayLocale,
+            'isActive' => $isActive,
+            'searchPhrase' => trim($searchPhrase),
+            'name' => $namesOut,
+        ];
+    }
+
+    /**
+     * Process the CSV row to store in DB
+     */
+    protected function processBatch(array $rows): void
+    {
+        try {
+            $values = [];
+            
+            foreach ($rows as $row) {
+                foreach ($row['name'] as $locale => $name) {
+                    $values[] = [
+                        'ror' => $row['ror'],
+                        'display_locale' => $row['displayLocale'],
+                        'is_active' => $row['isActive'],
+                        'search_phrase' => $row['searchPhrase'],
+                        'locale' => $locale,
+                        'setting_name' => 'name',
+                        'setting_value' => $name
+                    ];
+                }
+            }
+            DB::table($this->temporaryTable)->insert($values);
+
+            $values = DB::table($this->temporaryTable . ' as tmp')
+                ->select('tmp.ror', 'tmp.display_locale', 'tmp.is_active', 'tmp.search_phrase')
+                ->distinct()
+                ->leftJoin('rors as r', 'tmp.ror', '=', 'r.ror')
+                ->get()
+                ->map(function ($item) {
+                    return (array)$item;
+                })
+                ->all();
+            DB::table('rors')->upsert(
+                $values,
+                ['ror'],
+                ['display_locale', 'is_active', 'search_phrase']
+            );
+
+            $orphanedSettings = DB::table('ror_settings AS rs')
+                ->select('rs.ror_setting_id')
+                ->join('rors as r', 'rs.ror_id', '=', 'r.ror_id')
+                ->join($this->temporaryTable . ' as tmp1', 'tmp1.ror', '=', 'r.ror')
```

Comment:
It seems to me that this join isn't needed, I think the next join should be enough, just need to replace the first filter by `tmp2.ror = r.ror`, the `distinct()` shouldn't be needed as well.